sphinx/builders/html/__init__.py

```diff
@@ -1035,8 +1035,21 @@ def hasdoc(name: str) -> bool:
             templatename = newtmpl
 
         # sort JS/CSS before rendering HTML
```

Comment:
Have you considered doing something like
```suggestion
        # sort JS/CSS before rendering HTML
        ctx['script_files'] = list(ctx['script_files'])
        ctx['css_files'] = list(ctx['css_files'])

```
here?

Reply:
Acceptable.